app/uk/gov/hmrc/tariffclassificationfrontend/service/CasesService.scala

```diff
@@ -166,20 +167,24 @@ class CasesService @Inject()(appConfig: AppConfig,
     } yield updated
   }
 
-  def cancelRuling(original: Case, operator: Operator, clock: Clock = Clock.systemDefaultZone())
+  def cancelRuling(original: Case, reason: CancelReason,
+                   operator: Operator, clock: Clock = Clock.systemDefaultZone())
                   (implicit hc: HeaderCarrier): Future[Case] = {
     val updatedEndDate = LocalDate.now(clock).atStartOfDay(appConfig.zoneId)
 
     val decisionUpdating: Decision = original.decision
       .getOrElse(throw new IllegalArgumentException("Cannot Cancel a Case without a Decision"))
-      .copy(effectiveEndDate = Some(updatedEndDate.toInstant))
+      .copy(
+        effectiveEndDate = Some(updatedEndDate.toInstant),
+        cancellation = Some(Cancellation(reason = reason))
+      )
     val caseUpdating = original.copy(status = CaseStatus.CANCELLED, decision = Some(decisionUpdating))
 
     for {
       // Update the case
       updated: Case <- connector.updateCase(caseUpdating)
       // Create the event
-      _ <- addStatusChangeEvent(original, updated, operator)
+      _ <- addStatusChangeEvent(original, updated, operator, comment = Some(CancelReason.format(reason)))
```

Comment:
Similar point to previously. I think we should check with Joe if there is a requirement coming for us to add a Comment to a Cancellation. 
Otherwise we might need to stick this somewhere else, or concat it

It's better we find out now then have to do data migration in the future

Reply:
I think we also discussed there was a future reporting requirement where we will need to report on how many cases have cancellations with code `x`